### PR TITLE
Document :telemetry hookup

### PR DESCRIPTION
**Pull request by garthk**
Fixes the documentation gap behind #758, which is otherwise a WONTFIX as the system is behaving exactly as designed.

Could be better, but I'm trusting the review process to highlight which obvious areas of improvement need attention first. I'm most attracted to adding type information for the measurements and metadata, but maybe that's just me.

### CODE REVIEW

**Comment by garthk** [comment]
The lack of measurement and metadata information for the subscription events would be another obvious area for improvement.

File: guides/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+handler function to any of the following event names:
+
+- `[:absinthe, :execute, :operation, :start]` when the operation starts
+- `[:absinthe, :execute, :operation]` when the operation finishes
+- `[:absinthe, :subscription, :publish, :start]` when a subscription starts
+- `[:absinthe, :subscription, :publish]` when a subscription finishes
+- `[:absinthe, :resolve, :field, :start]` when field resolution starts
+- `[:absinthe, :resolve, :field]` when field resolution finishes
+
+Only fields with a resolver get measured.
```
Comment (benwilson512) [inline guides/telemetry.md]: Should probably be:
> By default, only fields with a resolver are measured.

File: guides/telemetry.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+For async, batch, and dataloader fields, Absinthe sends the final event when
+it gets the results. That might be later than when the results are ready. If
+you need to know how long the underlying operation took, you'll need to hook
+telemetry up to that underlying operation. See, for example, the recommended
+telemetry events in the documentaiton for `Ecto.Repo`.
+
+## Measurements
+
+- `start_time`: sent with event names ending `:start`
```
Comment (benwilson512) [inline guides/telemetry.md]: `ending` should maybe be `ending with` right?

File: guides/telemetry.md (hunk 1 above)
Comment (garthk) [reply guides/telemetry.md]: Oh, of course, I forgot people can add `meta: [absinthe_telemetry: true]` anywhere. Any other ways to override that default?

File: guides/telemetry.md (hunk 2 above)
Comment (garthk) [reply guides/telemetry.md]: Yep.

**Comment by garthk** [comment]
If any project contributors want to commit to this branch, feel free; I'll add `Co-authored-by` entries when I squash.

**Comment by benwilson512** [comment]
Thank you!

**Comment by garthk** [comment]
😂 please tell me you squashed out the `squash! squash!` markers

You're welcome!

**Pull request merged**